File: Python/Behavioral/State.py
class WritingState:
    def write(self, words):
        pass

class UpperCase(WritingState):
    def write(self, words):
        print words.upper()

class LowerCase(WritingState):
    def write(self, words):
        print words.lower()

class DefaultText(WritingState):
    def write(self, words):
        print words

class TextEditor():
    _state = None

    def __init__(self, state):
        self._state = state

    def setState(self, state):
        self._state = state

    def type(self, words):
        self._state.write(words)

editor = TextEditor(DefaultText())
editor.type('First Line')
editor.setState(UpperCase())

editor.type('Second Line')
editor.type('Third Line')

editor.setState(LowerCase())

editor.type('Fourth Line')
editor.type('Fifth Line')

File: examples-per-language/php/behavioral/State.php
<?php

use designPatternsForHumans\behavioral\State\DefaultState;
use designPatternsForHumans\behavioral\State\LowerCase;
use designPatternsForHumans\behavioral\State\TextEditor;
use designPatternsForHumans\behavioral\State\UpperCase;

require_once __DIR__ . '/autoload.php';

$editor = new TextEditor(new DefaultState());

$editor->type('First line');

$editor->setState(new UpperCase());

$editor->type('Second line');
$editor->type('Third line');

$editor->setState(new LowerCase());

$editor->type('Fourth line');
$editor->type('Fifth line');
